scipy/stats/tests/test_morestats.py

```diff
@@ -1669,36 +1673,38 @@ def test_permutation_method(self, size):
            12.10, 15.02, 16.83, 16.98, 19.92, 9.47, 11.68, 13.41, 15.35, 19.11]
 
 
+@array_api_compatible
 class TestKstat:
-    def test_moments_normal_distribution(self):
+    def test_moments_normal_distribution(self, xp):
         np.random.seed(32149)
-        data = np.random.randn(12345)
-        moments = [stats.kstat(data, n) for n in [1, 2, 3, 4]]
+        data = xp.asarray(np.random.randn(12345), dtype=xp.float64)
+        moments = xp.asarray([stats.kstat(data, n) for n in [1, 2, 3, 4]])
 
-        expected = [0.011315, 1.017931, 0.05811052, 0.0754134]
-        assert_allclose(moments, expected, rtol=1e-4)
+        expected = xp.asarray([0.011315, 1.017931, 0.05811052, 0.0754134],
+                              dtype=data.dtype)
+        xp_assert_close(moments, expected, rtol=1e-4)
 
         # test equivalence with `stats.moment`
         m1 = stats.moment(data, order=1)
         m2 = stats.moment(data, order=2)
         m3 = stats.moment(data, order=3)
-        assert_allclose((m1, m2, m3), expected[:-1], atol=0.02, rtol=1e-2)
+        xp_assert_close(xp.asarray((m1, m2, m3)), expected[:-1], atol=0.02, rtol=1e-2)
 
-    def test_empty_input(self):
+    def test_empty_input(self, xp):
         message = 'Data input must not be empty'
         with pytest.raises(ValueError, match=message):
-            stats.kstat([])
+            stats.kstat(xp.asarray([]))
 
-    def test_nan_input(self):
-        data = np.arange(10.)
-        data[6] = np.nan
+    def test_nan_input(self, xp):
+        data = xp.arange(10.)
+        data[6] = xp.nan
```

Comment:
Fine for now, but nice to save @lucascolley some trouble by generating these without mutation as we go forward. Is there a better way than?
```suggestion
        data = xp.where(data==6, xp.nan, data)
```

Reply:
Obvously here this is fine but I do wonder if we encountered mutations in the in loop of a solver, for example, if all the new allocations might be a bit painful performance wise

Reply:
Oh, I wouldn't change existing code like that all the time. Just easy cases like these tests, where it could make the difference between the test running with JAX or not.